django/contrib/admin/utils.py

```diff
@@ -401,7 +402,14 @@ def display_for_field(value, field, empty_value_display):
     from django.contrib.admin.templatetags.admin_list import _boolean_icon
 
     if getattr(field, "flatchoices", None):
-        return dict(field.flatchoices).get(value, empty_value_display)
+        flatchoices = field.flatchoices
+        if not isinstance(flatchoices, abc.Hashable) or not isinstance(
+            value, abc.Hashable
+        ):
+            flatchoices = make_hashable(flatchoices)
+            value = make_hashable(value)
+        return dict(flatchoices).get(value, empty_value_display)
```

Comment:
An alternative could be to try/catch, to avoid even more performance overhead from the condition:
```suggestion
        try:
            return dict(flatchoices).get(value, empty_value_display)
        except TypeError:
            flatchoices = make_hashable(flatchoices)
            value = make_hashable(value)
            return dict(flatchoices).get(value, empty_value_display)
```

Reply:
Hey @David-Wobrock — I think I prefer this `try...except` approach. (We're catching a pretty niche error no?) What do you think? 

(Looks good otherwise.)

Reply:
Agreed, and we avoid importing and testing against `collections.abc.Hashable` (which is currently never used in the Django code base).
I pushed edits :)